File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        reverse(s.begin(), s.end());
        int st = 0;
        int i = 0;
        string ans;
        while(i < s.size()){
            while(i<s.size() && s[i] == ' '){
                i++;
            }

            if(i>= s.size()) break;
            int st = i;
            while(i<s.size() && s[i] != ' '){
              i++;
            }
            int len = i - st;
            string word = s.substr(st,len);
            reverse(word.begin(), word.end());
            ans += word;
            ans += " ";
        }
        return ans.substr(0, ans.size()-1);
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
       s.trim();
      
       String[] arr= s.split(" ");
       StringBuilder ans = new StringBuilder();
       for(int i=arr.length-1;i>=0;i--){
        if(arr[i] != ""){
           ans.append(arr[i]);
             ans.append(" ");
        }
           
       }
       
       return ans.toString().substring(0,ans.length()-1).trim();
    }
}